Bogus CI fail:

```
2021-04-28T09:26:24.3177555Z Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not transfer artifact io.confluent:kafka-avro-serializer:jar:5.5.0 from/to confluent (https://packages.confluent.io/maven/): Transfer failed for https://packages.confluent.io/maven/io/confluent/kafka-avro-serializer/5.5.0/kafka-avro-serializer-5.5.0.jar
```

--- a/extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/PanacheEntityBase.java
+++ b/extensions/panache/hibernate-reactive-panache/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/PanacheEntityBase.java
@@ -703,6 +703,7 @@ public abstract class PanacheEntityBase {
      * @see #persist(Stream)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Iterable<?> entities) {
         return INSTANCE.persist(entities);
     }
@@ -716,6 +717,7 @@ public abstract class PanacheEntityBase {
      * @see #persist(Iterable)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Stream<?> entities) {
         return INSTANCE.persist(entities);
     }
@@ -729,6 +731,7 @@ public abstract class PanacheEntityBase {
      * @see #persist(Stream)
      * @see #persist(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Object firstEntity, Object... entities) {
         return INSTANCE.persist(firstEntity, entities);
     }

--- a/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheMongoEntityBase.java
+++ b/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/PanacheMongoEntityBase.java
@@ -772,6 +772,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persist(Stream)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persist(Iterable<?> entities) {
         INSTANCE.persist(entities);
     }
@@ -784,6 +785,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persist(Iterable)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persist(Stream<?> entities) {
         INSTANCE.persist(entities);
     }
@@ -796,6 +798,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persist(Stream)
      * @see #persist(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persist(Object firstEntity, Object... entities) {
         INSTANCE.persist(firstEntity, entities);
     }
@@ -808,6 +811,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #update(Stream)
      * @see #update(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void update(Iterable<?> entities) {
         INSTANCE.update(entities);
     }
@@ -820,6 +824,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #update(Iterable)
      * @see #update(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void update(Stream<?> entities) {
         INSTANCE.update(entities);
     }
@@ -832,6 +837,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #update(Stream)
      * @see #update(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void update(Object firstEntity, Object... entities) {
         INSTANCE.update(firstEntity, entities);
     }
@@ -844,6 +850,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persistOrUpdate(Stream)
      * @see #persistOrUpdate(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persistOrUpdate(Iterable<?> entities) {
         INSTANCE.persistOrUpdate(entities);
     }
@@ -856,6 +863,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persistOrUpdate(Iterable)
      * @see #persistOrUpdate(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persistOrUpdate(Stream<?> entities) {
         INSTANCE.persistOrUpdate(entities);
     }
@@ -868,6 +876,7 @@ public abstract class PanacheMongoEntityBase {
      * @see #persistOrUpdate(Stream)
      * @see #persistOrUpdate(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static void persistOrUpdate(Object firstEntity, Object... entities) {
         INSTANCE.persistOrUpdate(firstEntity, entities);
     }

--- a/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/reactive/ReactivePanacheMongoEntityBase.java
+++ b/extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/reactive/ReactivePanacheMongoEntityBase.java
@@ -785,6 +785,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persist(Stream)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Iterable<?> entities) {
         return INSTANCE.persist(entities);
     }
@@ -797,6 +798,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persist(Iterable)
      * @see #persist(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Stream<?> entities) {
         return INSTANCE.persist(entities);
     }
@@ -809,6 +811,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persist(Stream)
      * @see #persist(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persist(Object firstEntity, Object... entities) {
         return INSTANCE.persist(firstEntity, entities);
     }
@@ -821,6 +824,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #update(Stream)
      * @see #update(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> update(Iterable<?> entities) {
         return INSTANCE.update(entities);
     }
@@ -833,6 +837,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #update(Iterable)
      * @see #update(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> update(Stream<?> entities) {
         return INSTANCE.update(entities);
     }
@@ -845,6 +850,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #update(Stream)
      * @see #update(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> update(Object firstEntity, Object... entities) {
         return INSTANCE.update(firstEntity, entities);
     }
@@ -857,6 +863,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persistOrUpdate(Stream)
      * @see #persistOrUpdate(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persistOrUpdate(Iterable<?> entities) {
         return INSTANCE.persistOrUpdate(entities);
     }
@@ -869,6 +876,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persistOrUpdate(Iterable)
      * @see #persistOrUpdate(Object,Object...)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persistOrUpdate(Stream<?> entities) {
         return INSTANCE.persistOrUpdate(entities);
     }
@@ -881,6 +889,7 @@ public abstract class ReactivePanacheMongoEntityBase {
      * @see #persistOrUpdate(Stream)
      * @see #persistOrUpdate(Iterable)
      */
+    @GenerateBridge(callSuperMethod = true)
     public static Uni<Void> persistOrUpdate(Object firstEntity, Object... entities) {
         return INSTANCE.persistOrUpdate(firstEntity, entities);
     }

--- a/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/PanacheMockingTest.java
+++ b/integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/PanacheMockingTest.java
@@ -43,6 +43,9 @@ public class PanacheMockingTest {
         Assertions.assertSame(p, Person.findById(12l).await().indefinitely());
         Assertions.assertNull(Person.findById(42l).await().indefinitely());
 
+        Person.persist(p).await().indefinitely();
+        Assertions.assertNull(p.id);
+
         Mockito.when(Person.findById(12l)).thenThrow(new WebApplicationException());
         try {
             Person.findById(12l);
@@ -54,6 +57,7 @@ public class PanacheMockingTest {
         Assertions.assertTrue(Person.findOrdered().await().indefinitely().isEmpty());
 
         PanacheMock.verify(Person.class).findOrdered();
+        PanacheMock.verify(Person.class).persist(Mockito.<Object> any(), Mockito.<Object> any());
         PanacheMock.verify(Person.class, Mockito.atLeastOnce()).findById(Mockito.any());
         PanacheMock.verifyNoMoreInteractions(Person.class);
 
@@ -89,6 +93,9 @@ public class PanacheMockingTest {
         Assertions.assertSame(p, mockablePersonRepository.findById(12l).await().indefinitely());
         Assertions.assertNull(mockablePersonRepository.findById(42l).await().indefinitely());
 
+        mockablePersonRepository.persist(p).await().indefinitely();
+        Assertions.assertNull(p.id);
+
         Mockito.when(mockablePersonRepository.findById(12l)).thenThrow(new WebApplicationException());
         try {
             mockablePersonRepository.findById(12l);
@@ -101,6 +108,7 @@ public class PanacheMockingTest {
 
         Mockito.verify(mockablePersonRepository).findOrdered();
         Mockito.verify(mockablePersonRepository, Mockito.atLeastOnce()).findById(Mockito.any());
+        Mockito.verify(mockablePersonRepository).persist(Mockito.<Person> any());
         Mockito.verifyNoMoreInteractions(mockablePersonRepository);
     }
 

--- a/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/reactive/person/MockableReactivePersonRepository.java
+++ b/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/reactive/person/MockableReactivePersonRepository.java
@@ -1,0 +1,16 @@
+package io.quarkus.it.mongodb.panache.reactive.person;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoRepositoryBase;
+import io.quarkus.panache.common.Sort;
+import io.smallrye.mutiny.Uni;
+
+@ApplicationScoped
+public class MockableReactivePersonRepository implements ReactivePanacheMongoRepositoryBase<ReactivePersonEntity, Long> {
+    public Uni<List<ReactivePersonEntity>> findOrdered() {
+        return findAll(Sort.by("lastname", "firstname")).list();
+    }
+}

--- a/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/reactive/person/ReactivePersonEntity.java
+++ b/integration-tests/mongodb-panache/src/main/java/io/quarkus/it/mongodb/panache/reactive/person/ReactivePersonEntity.java
@@ -1,12 +1,20 @@
 package io.quarkus.it.mongodb.panache.reactive.person;
 
+import java.util.List;
+
 import org.bson.codecs.pojo.annotations.BsonId;
 
 import io.quarkus.mongodb.panache.reactive.ReactivePanacheMongoEntityBase;
+import io.quarkus.panache.common.Sort;
+import io.smallrye.mutiny.Uni;
 
 public class ReactivePersonEntity extends ReactivePanacheMongoEntityBase {
     @BsonId
     public Long id;
     public String firstname;
     public String lastname;
+
+    public static Uni<List<ReactivePersonEntity>> findOrdered() {
+        return findAll(Sort.by("lastname", "firstname")).list();
+    }
 }

--- a/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongodbPanacheMockingTest.java
+++ b/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/MongodbPanacheMockingTest.java
@@ -48,6 +48,9 @@ public class MongodbPanacheMockingTest {
         Assertions.assertSame(p, PersonEntity.findById(12L));
         Assertions.assertNull(PersonEntity.findById(42L));
 
+        PersonEntity.persist(p);
+        Assertions.assertNull(p.id);
+
         Mockito.when(PersonEntity.findById(12L)).thenThrow(new WebApplicationException());
         try {
             PersonEntity.findById(12L);
@@ -59,6 +62,7 @@ public class MongodbPanacheMockingTest {
         Assertions.assertTrue(PersonEntity.findOrdered().isEmpty());
 
         PanacheMock.verify(PersonEntity.class).findOrdered();
+        PanacheMock.verify(PersonEntity.class).persist(Mockito.<Object> any(), Mockito.<Object> any());
         PanacheMock.verify(PersonEntity.class, Mockito.atLeastOnce()).findById(Mockito.any());
         PanacheMock.verifyNoMoreInteractions(PersonEntity.class);
     }
@@ -92,6 +96,9 @@ public class MongodbPanacheMockingTest {
         Assertions.assertSame(p, mockablePersonRepository.findById(12L));
         Assertions.assertNull(mockablePersonRepository.findById(42L));
 
+        mockablePersonRepository.persist(p);
+        Assertions.assertNull(p.id);
+
         Mockito.when(mockablePersonRepository.findById(12L)).thenThrow(new WebApplicationException());
         try {
             mockablePersonRepository.findById(12L);
@@ -104,6 +111,7 @@ public class MongodbPanacheMockingTest {
 
         Mockito.verify(mockablePersonRepository).findOrdered();
         Mockito.verify(mockablePersonRepository, Mockito.atLeastOnce()).findById(Mockito.any());
+        Mockito.verify(mockablePersonRepository).persist(Mockito.<PersonEntity> any());
         Mockito.verifyNoMoreInteractions(mockablePersonRepository);
     }
 

--- a/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/reactive/ReactiveMongodbPanacheMockingTest.java
+++ b/integration-tests/mongodb-panache/src/test/java/io/quarkus/it/mongodb/panache/reactive/ReactiveMongodbPanacheMockingTest.java
@@ -1,0 +1,121 @@
+package io.quarkus.it.mongodb.panache.reactive;
+
+import java.time.Duration;
+import java.util.Collections;
+import java.util.List;
+
+import javax.ws.rs.WebApplicationException;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import io.quarkus.it.mongodb.panache.MongoTestResource;
+import io.quarkus.it.mongodb.panache.reactive.person.MockableReactivePersonRepository;
+import io.quarkus.it.mongodb.panache.reactive.person.ReactivePersonEntity;
+import io.quarkus.panache.mock.PanacheMock;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+import io.smallrye.mutiny.Uni;
+
+@QuarkusTest
+@QuarkusTestResource(MongoTestResource.class)
+public class ReactiveMongodbPanacheMockingTest {
+
+    private static final Duration timeout = Duration.ofSeconds(2);
+
+    @Test
+    @Order(1)
+    public void testPanacheMocking() {
+        PanacheMock.mock(ReactivePersonEntity.class);
+
+        Assertions.assertEquals(0, ReactivePersonEntity.count().await().atMost(timeout));
+
+        Uni<Long> uni23 = Uni.createFrom().item(23L);
+        Mockito.when(ReactivePersonEntity.count()).thenReturn(uni23);
+        Assertions.assertEquals(uni23, ReactivePersonEntity.count());
+
+        Uni<Long> uni42 = Uni.createFrom().item(42L);
+        Mockito.when(ReactivePersonEntity.count()).thenReturn(uni42);
+        Assertions.assertEquals(uni42, ReactivePersonEntity.count());
+
+        Mockito.when(ReactivePersonEntity.count()).thenCallRealMethod();
+        Assertions.assertEquals(0, ReactivePersonEntity.count().await().atMost(timeout));
+
+        PanacheMock.verify(ReactivePersonEntity.class, Mockito.times(4)).count();
+
+        ReactivePersonEntity p = new ReactivePersonEntity();
+
+        Uni<ReactivePersonEntity> uniP = Uni.createFrom().item(p);
+        Mockito.when(ReactivePersonEntity.<ReactivePersonEntity> findById(12L)).thenReturn(uniP);
+        Assertions.assertSame(uniP, ReactivePersonEntity.findById(12L));
+        Assertions.assertNull(ReactivePersonEntity.findById(42L).await().atMost(timeout));
+
+        ReactivePersonEntity.persist(p);
+        Assertions.assertNull(p.id);
+
+        Mockito.when(ReactivePersonEntity.findById(12L)).thenThrow(new WebApplicationException());
+        try {
+            ReactivePersonEntity.findById(12L);
+            Assertions.fail();
+        } catch (WebApplicationException x) {
+        }
+
+        Uni<List<ReactivePersonEntity>> uniEmptyList = Uni.createFrom().item(Collections.emptyList());
+        Mockito.when(ReactivePersonEntity.findOrdered()).thenReturn(uniEmptyList);
+        Assertions.assertEquals(uniEmptyList, ReactivePersonEntity.findOrdered());
+
+        PanacheMock.verify(ReactivePersonEntity.class).findOrdered();
+        PanacheMock.verify(ReactivePersonEntity.class).persist(Mockito.<Object> any(), Mockito.<Object> any());
+        PanacheMock.verify(ReactivePersonEntity.class, Mockito.atLeastOnce()).findById(Mockito.any());
+        PanacheMock.verifyNoMoreInteractions(ReactivePersonEntity.class);
+    }
+
+    @InjectMock
+    MockableReactivePersonRepository mockablePersonRepository;
+
+    @Test
+    public void testPanacheRepositoryMocking() throws Throwable {
+        Assertions.assertEquals(0, mockablePersonRepository.count().await().atMost(timeout));
+
+        Uni<Long> uni23 = Uni.createFrom().item(23L);
+        Mockito.when(mockablePersonRepository.count()).thenReturn(uni23);
+        Assertions.assertEquals(uni23, mockablePersonRepository.count());
+
+        Uni<Long> uni42 = Uni.createFrom().item(42L);
+        Mockito.when(mockablePersonRepository.count()).thenReturn(uni42);
+        Assertions.assertEquals(uni42, mockablePersonRepository.count());
+
+        Mockito.when(mockablePersonRepository.count()).thenCallRealMethod();
+        Assertions.assertEquals(0, mockablePersonRepository.count().await().atMost(timeout));
+
+        Mockito.verify(mockablePersonRepository, Mockito.times(4)).count();
+
+        ReactivePersonEntity p = new ReactivePersonEntity();
+        Uni<ReactivePersonEntity> uniP = Uni.createFrom().item(p);
+        Mockito.when(mockablePersonRepository.findById(12L)).thenReturn(uniP);
+        Assertions.assertSame(uniP, mockablePersonRepository.findById(12L));
+        Assertions.assertNull(mockablePersonRepository.findById(42L).await().atMost(timeout));
+
+        mockablePersonRepository.persist(p);
+        Assertions.assertNull(p.id);
+
+        Mockito.when(mockablePersonRepository.findById(12L)).thenThrow(new WebApplicationException());
+        try {
+            mockablePersonRepository.findById(12L);
+            Assertions.fail();
+        } catch (WebApplicationException x) {
+        }
+
+        Uni<List<ReactivePersonEntity>> uniEmptyList = Uni.createFrom().item(Collections.emptyList());
+        Mockito.when(mockablePersonRepository.findOrdered()).thenReturn(uniEmptyList);
+        Assertions.assertEquals(uniEmptyList, mockablePersonRepository.findOrdered());
+
+        Mockito.verify(mockablePersonRepository).findOrdered();
+        Mockito.verify(mockablePersonRepository, Mockito.atLeastOnce()).findById(Mockito.any());
+        Mockito.verify(mockablePersonRepository).persist(Mockito.<ReactivePersonEntity> any());
+        Mockito.verifyNoMoreInteractions(mockablePersonRepository);
+    }
+}

